### DISCUSSION TITLE
API Question

### DISCUSSION THREAD

**Discussion by AGoodGuy5543** [alvarobartt/investpy#323] [Q&A] [answered]
Hello alvarobartt,

A group of university students that I am currently working with have to choose a project to resolve some issues on for a final project. We decided to choose your project.
We are only solving the issues on our own branch, but we ran into an issue.
We chose issue #291 for our first issue and we're running in to a problem with the API - it's private.
In order for us to get more information from the website and the URL, we need to know how the API works and such.
Would it be possible to get some insight or access to the API, solely for private academia work? Any help in this matter would be greatly appreciated.

**Comment by alvarobartt** [comment, marked as answer]
Hi @AGoodGuy5543 I'm glad you chose `investpy` for your academic project! :hugs: 

Regarding the API there's no API reference as the API isn't public, but I got to understand how does it works via analyzing the web traffic when sending requests from the web. So I cannot provide you the API reference, but I can tell you that it's pretty simple, you just need to analyze the traffic while sending requests in the website, let's say, click `Income Statement` tab in https://www.investing.com/equities/apple-computer-inc-financial-summary and inspect the traffic so as to understand what the inner Investing.com API is doing! :muscle:

If you need anything else from my side, feel free to ask!

**Reply by Voila2000** [reply]
Hello alvarobartt,
Please tell me if the library works?
Yesterday everything worked, today it gives errors.
Investing.com changed the API ?

**Reply by alvarobartt** [reply]
It has already been solved, the new version will be released in a couple of minutes! Thank also both @nhlsm and @danilogalisteu, as the solution they proposed works like a charm.
